components/script/dom/htmlelement.rs

```diff
@@ -984,16 +988,19 @@ impl VirtualMethods for HTMLElement {
         // (The fact that it's enabled doesn't do much while it's
         // disconnected, but it is an observable fact to keep track of.)
         let element = self.as_element();
-        if self.is_form_associated_custom_element() && element.disabled_state() {
-            element.check_disabled_attribute();
-            element.check_ancestors_disabled_state_for_form_control();
-            if element.enabled_state() {
-                ScriptThread::enqueue_callback_reaction(
-                    element,
-                    CallbackReaction::FormDisabled(false),
-                    None,
-                );
+        if self.is_form_associated_custom_element() {
+            if element.disabled_state() {
+                element.check_disabled_attribute();
+                element.check_ancestors_disabled_state_for_form_control();
+                if element.enabled_state() {
+                    ScriptThread::enqueue_callback_reaction(
+                        element,
+                        CallbackReaction::FormDisabled(false),
+                        None,
+                    );
+                }
             }
+            element.check_readonly_attribute();
```

Comment:
It seems `unbind_from_tree` won't affect the "readonly" attribute, so maybe we could remove this change.